fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java

```diff
@@ -143,10 +165,60 @@ private Asset createAsset(InputStream in) throws IOException {
         return new ByteArrayAsset(str.toString().getBytes());
     }
 
+    private void prepareKeycloakMultitenancy() throws IOException {
+        if (keycloakMultitenancy != null && !keycloakMultitenancy.isEmpty()) {
+            // Prepare a relative paths to KC deployments map.
+            Map<String, KeycloakDeployment> deployments = new HashMap<>();
+            for (Map.Entry<String, String> entry : keycloakMultitenancy.entrySet()) {
+                InputStream is = getKeycloakJsonStream(entry.getValue());
+                if (is == null) {
+                    LOG.warn(String.format(
+                        "Unable to support the multitenancy due to the '%s' being not available", entry.getValue()
+                    ));
+                } else {
+                    deployments.put(entry.getKey(), KeycloakDeploymentBuilder.build(is));
+                }
+            }
+            try {
+                // Statically initialize KeycloakAdapterConfigResolver
+                Module module = Module.getBootModuleLoader().loadModule("org.wildfly.swarm.keycloak:deployment");
+                final String resolverClassName = "org.wildfly.swarm.keycloak.deployment.KeycloakAdapterConfigResolver";
+                Class<?> resolverClass = module.getClassLoader().loadClass(resolverClassName);
+                Method setDeployments = resolverClass.getDeclaredMethod("setDeployments", Map.class);
+                setDeployments.invoke(null, deployments);
```

Comment:
Whoah does it look weird. This so looks like it should be done elsewhere, but I really don't know where.

Reply:
This is how it is done in Swarm today